### ISSUE TITLE
[idea] Use vcpkg for dependencies?

### ISSUE THREAD

**Issue by jd28** [nwnxee/unified#620]
https://docs.microsoft.com/en-us/cpp/build/vcpkg?view=vs-2019

It's cross platform, simple, lots of packages.  It doesn't seem to support pinning to particular versions, but in the long run I still think it would be better than using system libraries.

**Comment by mtijanic** [comment]
As an entirely personal opinion, I'd like to avoid any dependencies for the core nwnx. Even cmake makes me really sad. If we could add this as an optional component to auto deal with your dependency, it'd be great - assuming the overhead of maintaining it is low.

In my ideal world, you'd be able to use env vars to set up all your deps manually and then just run `make`, and everything else comes as an optional value add.

I'm open to hearing a Reason You Suck speech though.

**Comment by jd28** [comment]
I agree core should avoid dependencies and should simplify simplify simplify (to quote Thoreau).  

I'll research it a bit more and how it could best fit in here.  Just from a brief look at it, it looks like a godsend for any C++ project.

There is at least one case: nwnx_redis where a dependency was included in tree and grows older and older.  Something like vcpkg could help avoid that.

**Comment by jd28** [comment]
And there will be no "Reason You Suck" speech coming from me.  You're a great person to lead this project.
